Python SDK topic-transactions docs (#16203)
ydb-platform/ydb: ydb/docs/en/core/reference/ydb-sdk/topic.md

  })
  ```

- Python

>   To write to a topic within a transaction, create a transactional writer by calling `topic_client.tx_writer` with the `tx` argument. Once created, you can send messages as usual. There's no need to close the transactional writer manually, as it will be closed automatically when the transaction ends.

Reviewer: Was there an example nearby of how to commit the transaction?

>   [Example on GitHub](https://github.com/ydb-platform/ydb-python-sdk/blob/main/examples/topic/topic_transactions_example.py)

Reviewer: 404

  ```python
  with ydb.QuerySessionPool(driver) as session_pool:

      def callee(tx: ydb.QueryTxContext):
          tx_writer = driver.topic_client.tx_writer(tx, topic)
          for i in range(message_count):
              tx_writer.write([ydb.TopicWriterMessage(data=str(i))])

>               print(f"Messages {messages} were written with tx.")

Reviewer: ```suggestion
              print(fMessages {messages} were written within a transaction successfully.")
```

>               print(f"Messages {messages} were written with tx.")

Reviewer: I don't see where `{messages}` has been defined.

      session_pool.retry_tx_sync(callee)
  ```

- Python (asyncio)

>   To write to a topic within a transaction, create a transactional writer by calling `topic_client.tx_writer` with the `tx` argument. Once created, you can send messages as usual. There's no need to close the transactional writer manually, as it will be closed automatically when the transaction ends.

Reviewer: ```suggestion
  To write to a topic within a transaction, create a transactional writer by calling `topic_client.tx_writer` with the `tx` argument. Once created, you can send messages as usual. There's no need to close the transactional writer manually, as it will be closed automatically when the transaction ends.
```

  })
  ```

- Python

  To write to a topic within a transaction, create a transactional writer by calling `topic_client.tx_writer` with the `tx` argument. Once created, you can send messages as usual. There's no need to close the transactional writer manually, as it will be closed automatically when the transaction ends.

>   In the example below, there is no explicit call to `tx.commit()` - it occurs implicitly upon successful completion of the lambda `callee`.

Reviewer: ```suggestion
  In the example below, there is no explicit call to `tx.commit()`; it occurs implicitly upon the successful execution of the `callee` lambda.
```

  [Example on GitHub](https://github.com/ydb-platform/ydb-python-sdk/blob/main/examples/topic/topic_transactions_example.py)

  ```python
  with ydb.QuerySessionPool(driver) as session_pool:

      def callee(tx: ydb.QueryTxContext):
          tx_writer: ydb.TopicTxWriter = driver.topic_client.tx_writer(tx, topic)

          for i in range(message_count):
              result_stream = tx.execute(query=f"select {i} as res;")
              for result_set in result_stream:
                  message = str(result_set.rows[0]["res"])
                  tx_writer.write(ydb.TopicWriterMessage(message))
                  print(f"Message {message} was written with tx.")

      session_pool.retry_tx_sync(callee)
  ```

- Python (asyncio)

  To write to a topic within a transaction, create a transactional writer by calling `topic_client.tx_writer` with the `tx` argument. Once created, you can send messages as usual. There's no need to close the transactional writer manually, as it will be closed automatically when the transaction ends.

>   In the example below, there is no explicit call to `tx.commit()` - it occurs implicitly upon successful completion of the lambda `callee`.

Reviewer: ```suggestion
  In the example below, there is no explicit call to `tx.commit()`; it occurs implicitly upon the successful execution of the `callee` lambda.
```

  }
  ```

- Python

>   To read messages from a topic within a transaction, use the `reader.recieve_batch_with_tx` method. It reads a batch of messages and adds their commit to the transaction, so there's no need to commit them separately. The reader can be reused across different transactions. However, it's important to commit transactions in the same order as the messages are read from the reader, as message commits in the topic must be performed strictly in order. The simplest way to ensure this is by using the reader within a loop.

Reviewer: ```suggestion
  To read messages from a topic within a transaction, use the `reader.receive_batch_with_tx` method. It reads a batch of messages and adds their commit to the transaction, so there's no need to commit them separately. The reader can be reused across different transactions. However, it's essential to commit transactions in the same order as the messages are read from the reader, as message commits in the topic must be performed strictly in order. The simplest way to ensure this is by using the reader within a loop.
```
In the example below it seems ok, so hopefully the mistype is only here.

>   To read messages from a topic within a transaction, use the `reader.recieve_batch_with_tx` method. It reads a batch of messages and adds their commit to the transaction, so there's no need to commit them separately. The reader can be reused across different transactions. However, it's important to commit transactions in the same order as the messages are read from the reader, as message commits in the topic must be performed strictly in order. The simplest way to ensure this is by using the reader within a loop.

Reviewer: Let's explicitly say what happens when the commits are not in order.

  [Example on GitHub](https://github.com/ydb-platform/ydb-python-sdk/blob/main/examples/topic/topic_transactions_example.py)

  ```python
  with driver.topic_client.reader(topic, consumer) as reader:
      with ydb.QuerySessionPool(driver) as session_pool:
          for _ in range(message_count):

              def callee(tx: ydb.QueryTxContext):
                  batch = reader.receive_batch_with_tx(tx, max_messages=1)
                  print(f"Messages [{batch.messages[0].data.decode()}] were read with tx.")

              session_pool.retry_tx_sync(callee)
  ```

- Python (asyncio)

>   To read messages from a topic within a transaction, use the `reader.recieve_batch_with_tx` method. It reads a batch of messages and adds their commit to the transaction, so there's no need to commit them separately. The reader can be reused across different transactions. However, it's important to commit transactions in the same order as the messages are read from the reader, as message commits in the topic must be performed strictly in order. The simplest way to ensure this is by using the reader within a loop.

Reviewer: ```suggestion
  To read messages from a topic within a transaction, use the `reader.receive_batch_with_tx` method. It reads a batch of messages and adds their commit to the transaction, so there's no need to commit them separately. The reader can be reused across different transactions. However, it's important to commit transactions in the same order as the messages are read from the reader, as message commits in the topic must be performed strictly in order. The simplest way to ensure this is by using the reader within a loop.
```
